workspaces/adventure-pack/goodies/python3/src/transpose_matrix/test.py

```diff
@@ -0,0 +1,51 @@
+from . import *
+import pytest
+
+
+def test_improper_input() -> None:
+    with pytest.raises(ValueError):
+        assert len(transpose_matrix([]))
+
+    with pytest.raises(ValueError):
+        assert len(transpose_matrix([[], []]))
+
+    with pytest.raises(ValueError):
+        assert len(transpose_matrix([[0], [0, 1]]))
+
+
+def assert_matrices_equal(list1, list2):
+    assert len(list1) == len(list2)
+    for row in range(len(list1)):
+        assert list1[row] == list2[row]
```

Comment:
I bet you can do `list == list2`, obviating the need for this function to exist.